tornado/test/web_test.py

```diff
@@ -183,6 +183,21 @@ def test_key_version_invalidate_version(self):
         self.assertEqual(new_handler.get_secure_cookie('foo'), None)
 
 
+class FinishTest(WebTestCase):
+    def get_handlers(self):
+        class FinishHandler(RequestHandler):
+            def get(self):
+                self.write("The finish method should return Future.")
+                f = self.finish()
+                assert isinstance(f, Future)
```

Comment:
There's a problem with this test: by the time you get to this point, you've already sent a 200 response code to the client, so this test can never fail. It would just log the error (That kind of works since we also count error logs as test failures, but failure reports are difficult to read in this case, so it's best to make a test that fails in an understandable way). 

I would suggest looking at IncorrectContentLengthTest for a way of passing information from the handler to the main body of the test after finish().